utils/build-script-impl

```diff
@@ -3287,7 +3287,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 ICU_BUILD_DIR=$(build_directory ${host} ${product})
                 ICU_INSTALL_DIR="$(get_host_install_destdir ${host})$(get_host_install_prefix ${host})"
                 ICU_LIBDIR="$(build_directory ${host} swift)/lib/swift/${SWIFT_HOST_VARIANT}/${SWIFT_HOST_VARIANT_ARCH}"
-                LIBICU_DEST_DIR="${ICU_INSTALL_DIR}lib/swift/${SWIFT_HOST_VARIANT}"
+                LIBICU_DEST_DIR="${ICU_INSTALL_DIR}lib/swift/${SWIFT_HOST_VARIANT}/${SWIFT_HOST_VARIANT_ARCH}"
```

Comment:
Should fix the libicu installation issue that cropped up on the linux CI, and have no effect on Darwin and Windows.